apps/api/src/course/course.resolver.ts

```diff
@@ -27,7 +37,139 @@ export class CourseResolver {
   async search(
     @Args('filter') filter: FilterInput,
     @Args('courseGroup') courseGroup: CourseGroupInput
-  ): Promise<Course[]> {
-    return this.courseService.search(filter, courseGroup)
+  ): Promise<ICourseSearchDocument[]> {
```

Comment:
This return type must match the type defined as `search` query return type in graphql schema.

Reply:
I see